controllers/dependencies.go

```diff
@@ -72,34 +71,83 @@ func (r *OperatorPipelineReconciler) reconcilePipelineDependencies(meta metav1.O
 			return err
 		}
 	}
+
 	return nil
 }
 
-func (r *OperatorPipelineReconciler) removePipelineDependencyFiles(filePath string) error {
-	if err := os.RemoveAll(filePath); err != nil {
-		log.Error(err, "Couldn't remove operator-pipelines directory")
+func (r *OperatorPipelineReconciler) applyPipelineManifests(fileName string, meta metav1.ObjectMeta) error {
+
+	b, err := os.ReadFile(fileName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Couldn't read manifest file for: %s", fileName))
 		return err
 	}
+
+	pipeline := &tekton.Pipeline{}
+
+	if err = yamlutil.Unmarshal(b, &pipeline); err != nil {
+		log.Error(err, fmt.Sprintf("Couldn't unmarshall yaml file for: %s", fileName))
+		return err
+	}
+
+	pipeline.SetNamespace(meta.Namespace)
+	err = r.Get(context.Background(), types.NamespacedName{Name: pipeline.Name, Namespace: pipeline.Namespace}, pipeline)
+
+	if len(pipeline.ObjectMeta.UID) > 0 {
+		if err := r.Client.Update(context.Background(), pipeline); err != nil {
+			log.Error(err, fmt.Sprintf("failed to update pipeline resource for file: %s", fileName))
+			return err
+		}
+	}
+
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+
+		if err := r.Client.Create(context.Background(), pipeline); err != nil {
+			log.Error(err, fmt.Sprintf("failed to create pipeline resource for file: %s", fileName))
+			return err
+		}
+	}
+
 	return nil
 }
 
-func (r *OperatorPipelineReconciler) applyManifests(fileName string, Namespace string, obj client.Object) error {
+func (r *OperatorPipelineReconciler) applyTaskManifests(fileName string, meta metav1.ObjectMeta) error {
 
 	b, err := os.ReadFile(fileName)
 	if err != nil {
-		log.Error(err, "Couldn't read manifest file")
+		log.Error(err, fmt.Sprintf("Couldn't read manifest file for: %s", fileName))
 		return err
 	}
 
-	if err = yamlutil.Unmarshal(b, &obj); err != nil {
-		log.Error(err, "Couldn't unmarshall yaml file")
+	task := &tekton.Task{}
```

Comment:
Nit: If you want a pointer, just use task := new(tekton.Task). Feel free to ignore. This gets the job done.

Reply:
I didn't know this was an option for structs, I changed to the more idiomatic approach. Thanks for pointing this out!